include/aspect/simulator_signals.h

```diff
@@ -250,6 +250,14 @@ namespace aspect
                                   const unsigned int compositional_index,
                                   const SolverControl &solver_control)> post_advection_solver;
 
+    /**
+    * A signal that is fired when the nonlinear solver scheme is done.
+    * The parameter is an object that contains information
+    * about the final state (failure/success), number of
+    * iterations and history of residuals of the nonlinear solver.
+    */
+    boost::signals2::signal<void (const SolverControl &)> post_nonlinear_solver;
```

Comment:
I think you want to document that this signal is also called for the linear schemes that do not actually iterate, and in that case will simply record a single iteration with a success outcome.